openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -481,24 +500,30 @@ def set_course_cohort_settings(course_key, **kwargs):
         cohorted_discussions (list): List of discussion ids.
 
     Returns:
-        A CourseCohortSettings object.
+        A CourseCohortSettings and CourseDiscussionSettings object.
 
     Raises:
         Http404 if course_key is invalid.
     """
     fields = {'is_cohorted': bool, 'always_cohort_inline_discussions': bool, 'cohorted_discussions': list}
-    course_cohort_settings = get_course_cohort_settings(course_key)
+    course_cohort_settings, course_discussion_settings = get_course_settings(course_key)
     for field, field_type in fields.items():
         if field in kwargs:
             if not isinstance(kwargs[field], field_type):
                 raise ValueError("Incorrect field type for `{}`. Type must be `{}`".format(field, field_type.__name__))
-            setattr(course_cohort_settings, field, kwargs[field])
-    course_cohort_settings.save()
-    return course_cohort_settings
+            if field == 'is_cohorted':
+                setattr(course_cohort_settings, field, kwargs[field])
+                course_cohort_settings.save()
+            elif field == 'always_cohort_inline_discussions':
+                setattr(course_discussion_settings, 'always_divide_inline_discussions', kwargs[field])
+            elif field == 'cohorted_discussions':
+                setattr(course_discussion_settings, 'divided_discussions', kwargs[field])
+    course_discussion_settings.save()
+    return course_cohort_settings, course_discussion_settings
 
 
 @request_cached
-def get_course_cohort_settings(course_key):
+def get_course_settings(course_key):
```

Comment:
Hmmm, I'd like to keep discussion settings distinct from cohort settings. I'll have to think about this.